tools/update-authors-json

```diff
@@ -38,76 +39,94 @@ parser.add_argument('--not-required', action='store_true', default=False,
                     help='Consider failures to reach GitHub nonfatal')
 args = parser.parse_args()
 
-def fetch_data(retries, link):
-    # type: (int, str) -> Optional[List[Dict[str, Any]]]
-    for _ in range(retries):
-        try:
-            r = requests.get(link)  # type: requests.Response
-            if r.status_code == 200:
-                return r.json()
-            else:
-                print('Github API return non 200 status.')
-        except requests.exceptions.RequestException as e:
-            print(e)
 
-    return None
+ContributorsJSON = TypedDict('ContributorsJSON', {
+    'date': str,
+    'contrib': List[Dict[str, Union[str, int]]],
+})
+
 
-def write_to_disk(json_data, out_file):
-    # type: (Dict[str, Any], str) -> None
+def fetch_contributors(repo_link: str) -> Optional[List[Dict[str, Dict[str, Any]]]]:
+    r = requests.get(repo_link)  # type: requests.Response
+    if r.status_code == 200:
+        return r.json()
+    else:
+        print("Github API returned {} when fetching {}".format(r.status_code, repo_link))
```

Comment:
The fact that we print this warning without any notice that it retried meant I got confused into thinking it had failed, when in fact it succeeded on retry... I think we should probably not print anything here (and just print an error if it fails all the retries).